lib/fetch.ts

```diff
@@ -1,21 +1,42 @@
-import { dummyServiceMap } from "./dummies.ts";
 import { client, meta } from "./grpc.ts";
+import { GetAllResponse } from "snitch-protos/protos/external.ts";
+import { PipelineInfo } from "snitch-protos/protos/info.ts";
+import { dummyServiceMap } from "./dummies.ts";
+import {
+  FlowEdge,
+  FlowNode,
+  mapAudiencePipelines,
+  mapEdges,
+  mapNodes,
+} from "../components/serviceMap/customNodes.tsx";
+
+export type ServiceMapType = GetAllResponse & { pipes: PipelineInfo[] };
+
+export const getServiceMap = async (): Promise<ServiceMapType> => {
+  // const { response } = await client.getAll({}, meta);
+  return {
+    ...dummyServiceMap,
+    pipes: Object.values(dummyServiceMap?.pipelines),
+  };
+};
 
-export const getServiceMap = async () => {
-  try {
-    const { response } = await client.getServiceMap({}, meta);
+export type ServiceNodes = { nodes: FlowNode[]; edges: FlowEdge[] };
 
-    return response;
-  } catch (error) {
-    console.error("error fetching service map", error);
-    console.log("returning dummy data instead");
-    return dummyServiceMap;
-  }
+export const getServiceNodes = async (): Promise<ServiceNodes> => {
```

Comment:
Pushed the node mapping down into the fetch layer so we can call it from everywhere in the server side handlers. I think I've updated all the handlers to use this.